CMakeLists.txt

```diff
@@ -554,14 +558,10 @@ if(FLATBUFFERS_BUILD_TESTS)
   add_executable(flatsampletext ${FlatBuffers_Sample_Text_SRCS})
   add_executable(flatsamplebfbs ${FlatBuffers_Sample_BFBS_SRCS})
 
-  # Add a library so there is a single target that the generated samples can 
-  # link too.
-  add_library(flatsample INTERFACE)
-
-  # Since flatsample has no sources, we have to explicitly set the linker lang.
-  set_target_properties(flatsample PROPERTIES LINKER_LANGUAGE CXX)
-  
-  compile_schema_for_samples(samples/monster.fbs "${FLATC_OPT_COMP}")
+  flatbuffers_generate_headers(TARGET flatsample
```

Comment:
Minor change: CMakeLists.txt now calls the flatbuffers_generate_headers() function at least once.  That way, if there are any issues with this function, they will be caught by CI